packages/itwin/grouping-mapping-widget/src/widget/components/GroupsShowHideButtons.tsx

```diff
@@ -0,0 +1,48 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import type { Group } from "@itwin/insights-client";
+import { SvgVisibilityHide, SvgVisibilityShow } from "@itwin/itwinui-icons-react";
+import { IconButton } from "@itwin/itwinui-react";
+import React, { useCallback, useMemo } from "react";
+
+interface GroupsShowHideButtonsProps {
+  group: Group;
+  hiddenGroupsIds: string[];
+  isLoadingQuery: boolean;
+  setHiddenGroupsIds: (ids: string[]) => void;
```

Comment:
It might be better to use a `Set` for this to ensure uniqueness. `includes` operations should work a bit faster with sets.

Reply:
I agree, changed.